### ISSUE TITLE
Initialization action documentation is out-of-date

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#3434]
Much of it hasn't been kept up-to-date as the actions themselves have changed.

**Comment by nilsvu** [comment]
It might be easier to just look at the `simple_tags` and `compute_tags` lists than keeping the docs up-to-date. Since `SetupDataBox` was introduced the init actions are a lot more self-documenting than before.

**Comment by wthrowe** [comment]
I agree in general.  The reason this came up now was that we need documentation and rationale for adding mutable cache tags to the DataBox.
